### ISSUE TITLE
Support 'Range between' along with 'rows between' in `window_frame`

### ISSUE THREAD

**Issue by talegari** [tidyverse/dbplyr#1435]
Right now, `dbplyr::window_frame` checks if `from` and `to` are numbers.

```r
  check_number_whole(from, allow_infinite = TRUE)
  check_number_whole(to, allow_infinite = TRUE)
```

It would be great to have support interval arguments for `from/to` so that `dbplyr::window_frame` will support 'Range between' queries. Here is an example from duckdb's window functions [page](https://duckdb.org/docs/sql/window_functions.html):

```sql
SELECT "Plant", "Date",
    min("MWh") OVER seven AS "MWh 7-day Moving Minimum",
    avg("MWh") OVER seven AS "MWh 7-day Moving Average",
    max("MWh") OVER seven AS "MWh 7-day Moving Maximum"
FROM "Generation History"
WINDOW seven AS (
    PARTITION BY "Plant"
    ORDER BY "Date" ASC
    RANGE BETWEEN INTERVAL 3 DAYS PRECEDING
              AND INTERVAL 3 DAYS FOLLOWING)
ORDER BY 1, 2;
```

**Comment by hadley** [comment]
I don't think this is a simple change to `window_frame()` because you can also use numeric values in `RANGE` mode. So this would likely need to either be a separate function or we'd need to add a `mode` argument to `window_frame()`.

* Duckdb docs: https://duckdb.org/docs/sql/window_functions#range-framing
* PostgreSQL docs: https://www.postgresql.org/docs/current/sql-expressions.html#SYNTAX-WINDOW-FUNCTIONS. Note example of `RANGE BETWEEN '1 day' PRECEDING AND '10 days' FOLLOWING` so uses a different syntax to duckdb.

**Comment by hadley** [comment]
I think this is probably out of scope given that dplyr doesn't support range queries.
